### DISCUSSION TITLE
AOT error when deserialising on a Il2CPP mac build

### DISCUSSION THREAD

**Discussion by paatz04** [applejag/Newtonsoft.Json-for-Unity#67] [Q&A] [answered]
Hi,

So I'm getting the following error when I'm trying to deserialise a string in a mac Il2CPP build. It used to work but after doing a single windows build it doesn't anymore. It still works when doing a mono build. 

`ExecutionEngineException: Attempting to call method 'Newtonsoft.Json.Utilities.CollectionWrapper`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]::.ctor' for which no ahead of time (AOT) code was generated.`

I'm using the latest github version 12.0.301

Any idea what this could be ?
Thank you

**Comment by paatz04** [comment, marked as answer]
Sorry, solved by adding:

```
AotHelper.EnsureList<int>();
AotHelper.EnsureType<CustomType>();
AotHelper.EnsureType<HashSet<int>>();
AotHelper.EnsureType<HashSet<string>>();
```

**Comment by applejag** [comment]
Hehe don't be sorry! This project is in need of providing the info on using the AotHelper more clearly.

Thanks for reporting this, and by finding a solution. :) It helps others find solutions a lot
